### PR TITLE
Audio recorder fragment

### PR DESCRIPTION
**Pull request by djordjemarusic**

### CODE REVIEW

File: mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/MicFragment.kt [hunk 1]
```diff
@@ -1,17 +1,80 @@
...
+    private val UPDATE_SPACE_TIME_MS: Long = 60000
+    private var lastUpdateTime: Long = 0
+
+    // handling MediaFile
+    private var handlingMediaFile: VaultFile? = null
+
+    // recording
+    private var uploadPresenter: TellaFileUploadSchedulePresenter? = null
+    private val disposable = CompositeDisposable()
+    var presenter: AudioCapturePresenter? = null
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/MicFragment.kt:66]: `uploadPresenter`, `presenter`, `audioRecorder` could be initialized as **lazy** like this 

`private val audioRecorder  by lazy { AudioCapturePresenter(this)  }` like this you are sure that the object will be created only when it is called, and you will always use the same object in the current scope.

File: mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/MicFragment.kt [hunk 2]
```diff
@@ -1,17 +1,80 @@
...
+    lateinit var mRecord: ImageButton
+    lateinit var mPlay: ImageButton
+    lateinit var mStop: ImageButton
+    lateinit var mTimer: TextView
+    lateinit var freeSpace: TextView
+    lateinit var redDot: ImageView
+    lateinit var recordingName: TextView
+
     private var param1: String? = null
     private var param2: String? = null
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/MicFragment.kt:79]: what is the use of this params ?

File: mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/MicFragment.kt [hunk 3]
```diff
@@ -41,11 +158,308 @@ class MicFragment : Fragment() {
...
+    }
+
+    override fun onStop() {
+        activity.stopLocationMetadataListening()
+
+        super.onStop()
+    }
+
+    override fun onDestroy() {
+        animator!!.end()
```
Comment (ahlem-jarrar) [inline mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/MicFragment.kt:181]: app can crash here if animator is null, maybe use null safe check `animator?.end()`

File: mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/MicFragment.kt (hunk 2 above)
Comment (djordjemarusic) [reply mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/MicFragment.kt:79]: True, don't need it. In fact, I just need one for animation.

**Review by ahlem-jarrar — APPROVED** [review]

**Pull request merged**